### ISSUE TITLE
clarify between multiple reshares

### ISSUE THREAD

**Issue by BEG65** [diaspora/diaspora#7967]
Currently, when I reshare a post I get notifications for it along the lines of "Reshare by (me)".

If I reshare more than one post, they are ALL called "Reshare by (me)".

This makes the notification list completely incomprehensible. I'd like to see the original title of the original post be retained in the "Reshare..." notification. So that I'd get something like

"(person) liked your reshare of (original title)" and "(person) commented on your reshare of (original title)"

As it is, I wind up posting links to items I would otherwise reshare in order to get some sanity in my notification list. 

I looked through the issues and googled around for any potential discussion of this, but if there's any, I missed it. Thanks!

**Comment by Flaburgan** [comment]
Thank you for opening an issue @BEG65, that's indeed a very good suggestion!

**Comment by goobertron** [comment]
This looks the same as the long-standing #5066 – feel free to add your suggestions there!

**Comment by davidmoseler** [comment]
I am working on this one. The main feature is basically implemented. There is a problem though. What should happen if the original post is deleted?

There is a spec (which I broke) that reads "falls back to the title, if the root post was deleted". But the spec implementation doesn't make much sense, because, before, the text was only "Reshare by ${author}", thus not mentioning the deleted root post, and nothing changed before. Now that I made a change, I need to update de spec.

So, which behaviour would be the best one when the post root is deleted?

**Comment by BEG65** [comment]
Obviously I can't speak for everyone, but myself I would be satisfied if original post title lost its link upon deletion. It could also change to say "reshare of (post deleted)" directly. I don't know that this happens terribly often, so I think its fine if the link to the original post goes away. 

What I'm not clear on is whether a reshared post is a copy of the original post or if it's actually the original post, but showing only the likes and comments generated by my reshare of it? If the original post is deleted then I think it's fine if the notification says something like "reshare of (deleted post)".

If I'm sharing multiple posts that all get deleted, t here's some other issue going on at that point ;-)

**Comment by BEG65** [comment]
Goobertron, issue #5066 deals with reshares of a reshare... Seems to be slightly different (but yes related). Thank you!

**Comment by MasterOfTheTiger** [comment]
@BEG65 
> Goobertron, issue #5066 deals with reshares of a reshare... Seems to be slightly different (but yes related). Thank you!

Yeah, these look like duplicates. At least they are extremely similar, and ask for similar solution. The [second comment](https://github.com/diaspora/diaspora/issues/5066#issuecomment-48626538) seems to address this issue particularly

**Comment by BEG65** [comment]
Mostly I want the original p ost title in there so I can distinguish among a reshare of A and a reshare of B. If it's deleted then fine put in a generic "deleted psot". If it's a reshare of a reshare of a reshare or whatever, just so long as it's unique from the /other/ posts I've reshared, I'll be happy. But right now it's so generic that if I start getting notifications on two or three reshares I've done I'm hopelessly lost...

**Comment by goobertron** [comment]
> Goobertron, issue #5066 deals with reshares of a reshare... Seems to be slightly different (but yes related). Thank you!

It deals with all interactions of a reshare (see the first line), like this issue; but I noted that the wording of the notification when someone reshares a reshare can seem a bit ugly. Some care is needed over the wording in that case, and that's something that should be discussed, perhaps in Discourse.

@davidmoseler, thank you for working on this. Please have a look at the earlier issue, #5066, for some more suggestions how to deal with this.

**Comment by BEG65** [comment]
If it is important to indicate a reshare is of an original or another reshare, perhaps that could be shortened to a ... (to fit in the notification drop down) that can be expanded to show the path: 
Reshare by (user1) of the reshare by (user2) ... of the original post (title, by userN)

Or something like that?

[I personally would not care; but I understand there's all the edge cases that need to be listed & accounted for.]

**Comment by goobertron** [comment]
I believe that a reshare is now linked back directly to the original post, even if you reshare from someone else's reshare of that post. So it wouldn't be possible to say 'reshare B of reshare A of original post'. And in any case, it's long and complicated to add to the notifications.

Can we close this issue and reach a solution on #5066, where I suggested wordings for the various interactions on a reshare? It's best to keep only one ticket open per issue.

**Comment by davidmoseler** [comment]
"Reshare by (user1) of the reshare by (user2) ... of the original post (title, by userN)"

This doesn't look to me as a scalable solution. In facebook, people reach a "stack depth" of reshares counting in the thousands, as they reshare reshares of reshares of (...) of a reshared post. Theoretically any number of iteractions should be feasible, but this is certainly not the case when we are concerned with the size of the notification phrase.

**Comment by BEG65** [comment]
I totally agree. I think "Reshare of (original title)" (or original deleted if that's the case) regardless of the depth is perfectly fine. The only reason I suggested it is that #5066 is about "Reshare of reshare..." -- which I could care less about, but seems to be important somehow.

**Comment by goobertron** [comment]
@BEG65 

> The only reason I suggested it is that #5066 is about "Reshare of reshare..."

No it isn't. Read the first line:

> I'd like to suggest a tweak to the notification when someone interacts with (***likes, comments on or reshares***) a post you have reshared from someone else.

It is exactly what you are proposing here: that, instead of saying, for instance, 'X has liked your post Reshare by BEG65', a notification would read 'X has liked your reshare We need more developers' (in my original example). 

I explicitly mentioned the case of someone resharing your reshare because the wording 'X has reshared your reshare We need more developers' is a little clumsy ('reshared your reshare'), and could probably be improved. But the issue concerns notifications for all interactions on a reshared post.

Please read #5066, and you will see. As I said, this issue is a duplicate, and should be closed so that any necessary discussion can take place on the original ticket.

**Comment by BEG65** [comment]
So many examples.
> David also commented on Davids post Reshare by David.
And
> Joe has liked your post Reshare by Cindy. 
> Jill has liked your post Reshare by Cindy. 
> Chris has liked your post Reshare by Cindy. 

where in fact each of those is a reshare of a **different** post.

I can move this over to #5066 (hey I'm the noob here) but there's more examples than the one you provided, which IS a reshare of a reshare example, which is what makes it appear to be concerned about the nested reshare. I look closely at examples, years of debugging people's complaints has taught me to look at the examples of the problem they're having.
